cuppa-core/src/test/java/org/forgerock/cuppa/TaggedTests.java

```diff
@@ -0,0 +1,468 @@
+/*
+ * Copyright 2016 ForgeRock AS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.forgerock.cuppa;
+
+import static org.forgerock.cuppa.Cuppa.*;
+import static org.forgerock.cuppa.Cuppa.when;
+import static org.forgerock.cuppa.CuppaTestProvider.runTests;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.forgerock.cuppa.functions.TestFunction;
+import org.forgerock.cuppa.internal.TestContainer;
+import org.forgerock.cuppa.model.Tags;
+import org.forgerock.cuppa.reporters.Reporter;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class TaggedTests {
+
+    @BeforeMethod
+    public void setup() {
+        TestContainer.INSTANCE.reset();
+        TestContainer.INSTANCE.setTestClass(HookTests.class);
+    }
+
+    @Test
+    public void shouldRunAllTestsWithNoRunTagsSpecified() throws Exception {
+
+        //Given
+        Reporter reporter = mock(Reporter.class);
+        TestFunction testFunction = mock(TestFunction.class);
+        {
+            describe("tagged tests", () -> {
+                it("runs the tagged test")
+                        .withTags("smoke")
+                        .asserts(testFunction);
+            });
+        }
+
+        //When
+        runTests(reporter);
+
+        //Then
+        verify(testFunction).apply();
+        verify(reporter).testPass(any());
+    }
+
+    @Test
+    public void shouldBeAbleToTagATest() throws Exception {
```

Comment:
After reading the rest of the tests, I think this test and the one below are redundant. I'd just remove them.